### PR TITLE
feat(connector): unified csv parser

### PR DESCRIPTION
**Pull request by idx0-dev**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We now have a unified CSV parser implementation for our source connector, which can be used by both the message-based (Kafka) and file-based (S3) connectors, thanks to the NdByteStreamWrapper. With this implementation, we can now support the CSV format in our message-based stream.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by waruto210 — COMMENTED** [review]
Good job, please fix the ci.

File: ci/workflows/pull-request.yml [hunk 1]
```diff
@@ -43,270 +43,18 @@ steps:
...
           environment:
-            - GITHUB_TOKEN
-    timeout_in_minutes: 10
-    retry: *auto-retry
-
-  - label: "build (deterministic simulation)"
-    command: "ci/scripts/build-simulation.sh"
-    key: "build-simulation"
-    plugins:
-      - gencer/cache#v2.4.10: *cargo-cache
```
Comment (tabVersion) [inline ci/workflows/pull-request.yml:66]: deleted workflow?

**Review by tabVersion — COMMENTED** [review]
parser part LGTM
please restore the ci

File: ci/workflows/pull-request.yml (hunk 1 above)
Comment (idx0-dev) [reply ci/workflows/pull-request.yml:66]: Just for testing my new workflow, without waiting others. I will move it to main-cron once it becomes green.

**Comment by idx0-dev** [comment]
https://buildkite.com/risingwavelabs/main-cron/builds/375#0186cb66-67d5-430a-b0df-f61ea10c959b

**Pull request merged**